### PR TITLE
pythonPackages.fs-s3fs: init at 1.0.0

### PR DESCRIPTION
**Pull request by Ma27**
###### Motivation for this change

`fs` is a python-based file system abstraction layer. The new package
`fs-s3fs` is an implementation of it which stores files inside an S3
bucket.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Ma27** [comment]
cc @ironpinguin

**Comment by Ma27** [comment]
@GrahamcOfBorg build pythonPackages.fs-s3fs python3Packages.fs-s3fs

**Comment by Ma27** [comment]
rebuilding due to timeouts (I guess that's been an issue with borg as the package is actually trivial)

@GrahamcOfBorg build pythonPackages.fs-s3fs python3Packages.fs-s3fs

**Comment by Ma27** [comment]
The original problem seems to be `pyftpdlib` which has been packaged some months ago and seems to break on Hydra. Currently investigating....

**Comment by Ma27** [comment]
I decided to skip the tests in `pyftpdlib` for now. I'm not that surprised that an FTP library has a certain set of impure tests (that behave differently in Hydra builds and Borg tests, locally the simply worked).

File: pkgs/development/python-modules/fs-s3fs/default.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  pname = "fs-s3fs";
+  version = "1.0.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1czv67zs4sl5l3rv9l3hzn22zzzqm372lq1wlhihigir4cfyslak";
+  };
+
+  propagatedBuildInputs = [ fs six boto3 ];
+  checkInputs = [ nose ];
```
Comment (worldofpeace) [inline pkgs/development/python-modules/fs-s3fs/default.nix]: I'm not seeing the test being ran in the output `Ran 0 tests in 0.118s`

**Comment by Ma27** [comment]
Thanks for the reminder! I decided to skip the tests as they heavily rely on s3 buckets and such tests arehighly impure.

**Comment by worldofpeace** [comment]
> Thanks for the reminder! I decided to skip the tests as they heavily rely on s3 buckets and such tests arehighly impure.

All these impurities and we'll need a software filter :smile: 

I'm testing locally. Should be good though.

**Review by worldofpeace — APPROVED** [review]
Great, merging at will.

Also :+1: for your detailed commit msg for `pythonPackages.pyftpdlib`.

**Pull request merged**